app/Semester.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace App;
+
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Model;
+
+/** A semester is identified by a year and by it's either autumn or spring.
+ * ie. a spring semester starting in february 2020 will be (2019, 2) since we write 2019/20/2.
+ * The autumn semester starting in september 2020 is (2020, 1) since we write 2020/21/1.
+ */
+class Semester extends Model
+{
+    protected $table = 'semesters';
+    protected $primaryKey = 'id';
+    public $incrementing = true;
+    public $timestamps = false;
+    protected $fillable = [
+        'year',
+        'part',
+    ];
+
+    private const SEPARATOR = '-';
+
+    const PARTS = [1, 2];
+    const ACTIVE = 'ACTIVE';
+    const INACTIVE = 'INACTIVE';
+    const PASSIVE = 'PASSIVE';
+    const PENDING = 'PENDING';
+    const STATUSES = [
+        self::ACTIVE,
+        self::INACTIVE,
+        self::PASSIVE,
+        self::PENDING,
+    ];
+
+    // Values are in month
+    // TODO: change to dates?
+    const START_OF_SPRING_SEMESTER = 2;
+    const END_OF_SPRING_SEMESTER = 7;
+    const START_OF_AUTUMN_SEMESTER = 8;
+    const END_OF_AUTUMN_SEMESTER = 1;
+
+    public function tag()
+    {
+        return $this->year.self::SEPARATOR.($this->year + 1).self::SEPARATOR.$this->part;
+    }
+
+    public function isAutumn()
+    {
+        return $this->part == 1;
+    }
+
+    public function isSpring()
+    {
+        return $this->part == 2;
+    }
+
+    public function users()
+    {
+        return $this->belongsToMany(User::class, 'semester_status')->withPivot(['status', 'comment']);
+    }
+
+    public function usersWithStatus($status)
+    {
+        return $this->belongsToMany(User::class, 'semester_status')
+                    ->wherePivot('status', '=', $status)
+                    ->withPivot('comment');
+    }
+
+    public function activeUsers()
+    {
+        return $this->usersWithStatus(self::ACTIVE);
+    }
+
+    public function hasUserWith($user, $status)
+    {
+        return $this->usersWithStatus($status)->get()->contains($user);
+    }
+
+    public function isActive($user)
+    {
+        return $this->hasUserWith($user, self::ACTIVE);
+    }
+
+    public static function newest()
+    {
+        return Semester::orderBy('year', 'desc')->orderBy('part', 'desc')->first();
+    }
+
+    // There is always a "current" semester. If there is not in the database, this function creates it.
+    // TODO: fine a safer method?
```

Comment:
Note that I did not use composite keys here because Eloquent does not support it (even though that would be the safest).